certified-mail-sender/src/main/scala/it/pagopa/interop/certifiedMailSender/Main.scala

```diff
@@ -20,12 +19,12 @@ object Main extends App {
   val sqsHandler: SQSHandler = SQSHandler(ApplicationConfiguration.queueUrl)(blockingEC)
   val job: JobExecution      = JobExecution(sqsHandler)
 
-  def execution: Future[Unit] = job
+  def execute(): Future[Unit] = job
     .run()
+    .flatMap(_ => execute())
     .recover(ex => logger.error("There was an error while running the job", ex))
     .andThen(_ => blockingThreadPool.shutdown())(global)
 
-  Await.result(execution, Duration.Inf)
-  logger.info("Completed certified mails sender job")
+  execute()
```

Comment:
What is waiting for completion here? doesn't the job this way complete immediately?

Reply:
+1

Reply:
Apart from the previous comment, if you have a better solution to implement a long-run job with Future is welcome.

Reply:
Fine by me, I would just leave the completion log to be sure that the pod did not crash during the execution

Reply:
Hmm, awaiting will probably block a thread, so it's not ideal in this case. The job will survive as long as the Execution Context this future is scheduled on is not composed only of daemon threads (but it's not this case).
Also, since the scope you're calling this `execute()` in is the `Main` scope, I expect something like an implicit await from the main thread to happen, so PROBABLY an await is happening here nevertheless.

Reply:
TLDR: this snippet is fine